homeassistant/components/intellifire/__init__.py

```diff
@@ -39,3 +54,24 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+# Example migration function
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    version = config_entry.version
+
+    if version == 1:
+
+        # Version 1 doesn't have API information so we will replace username/password with dummy data and a reauth flow will get triggered above
+
+        config_entry.version = 2
+        new = {**config_entry.data}
+        new[CONF_USERNAME] = ""
+        new[CONF_PASSWORD] = ""
```

Comment:
Why do you need to nullify the user/pass and then ask it again to login to the iftapi website ?